tests/installation/validate_zfcp_multipath.pm

```diff
@@ -0,0 +1,84 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Test module to validate multipath on z/VM testing infrastructure.
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use Test::Assert ':all';
+
+sub run {
+    select_console 'install-shell';
+    verify_host_bus_adapters_attached();
+    verify_plain_scsi_block_devices();
+    verify_multipath_block_devices();
+    select_console 'installation';
+}
+
+#Verify that there are two Host Bus adapters Attached (HBA), and that the corresponding channels (0.0.fa00, 0.0.fc00) are listed
+sub verify_host_bus_adapters_attached() {
+    my @EXPECTED_HBA = ("", "0.0.fa00", "0.0.fc00");
+    my @actual_hba = split /\n/, script_output "ls -l /sys/class/fc_host";
+    # output must have 3 lines, last two have useful info
+    assert_equals 3, @actual_hba, "Invalid number of host bus adapters";
+    for my $i (1 .. 2) {
+        assert_matches qr/$EXPECTED_HBA[$i]/, $actual_hba[$i], "Channel not found: $EXPECTED_HBA[$i]";
+    }
+}
+
+#verify that there are 8 SCSI block devices listed (as a result of having two adapters, two hard disks and two paths to each disk, 2^3 = 8)
+#verify that LUN 0x4001403200000000 corresponds to the 214GB disk, and LUN 0x4001405000000000 to the 42.9GB disk
+sub verify_plain_scsi_block_devices() {
+    my @all_scsi_list = split /\n/, script_output "lsscsi -xxgst";
+    # must have a total of 8 devices listed
+    assert_equals 8, @all_scsi_list, "Expected 8 SCSI block devices";
+    assert_number_scsi_block_devices(4, "0x4001405000000000", "42.9GB", @all_scsi_list);
```

Comment:
you don't need an extra function to do this, as there are few items is efficient to traverse the list twice and also more maintainable. Please check this snippet:
```my @all_scsi_list = split /\n/, $line;
my @small_disk = grep {/0x4001405000000000.*42.9GB/} @all_scsi_list;
my @big_disk = grep {/0x4001403200000000.*214GB/} @all_scsi_list; 
print scalar @small_disk . "\n";
print scalar @big_disk . "\n";
 ```